src/components/track-tile/TrackTile.tsx

```diff
@@ -1,106 +1,215 @@
+import { User } from 'audius-client/src/common/models/User'
+import { getUserId } from 'audius-client/src/common/store/account/selectors'
+import { getTrack } from 'audius-client/src/common/store/cache/tracks/selectors'
+import { getUserFromTrack } from 'audius-client/src/common/store/cache/users/selectors'
+import {
+  repostTrack,
+  saveTrack,
+  undoRepostTrack,
+  unsaveTrack
+} from 'audius-client/src/common/store/social/tracks/actions'
+import {
+  OverflowAction,
+  OverflowSource
+} from 'audius-client/src/common/store/ui/mobile-overflow-menu/types'
+import { requestOpen as requestOpenShareModal } from 'audius-client/src/common/store/ui/share-modal/slice'
+import { open as openOverflowMenu } from 'common/store/ui/mobile-overflow-menu/slice'
+import { isEqual } from 'lodash'
+import {
+  Animated,
+  Easing,
+  GestureResponderEvent,
+  Pressable,
+  StyleSheet
+} from 'react-native'
+import { useDispatch, useSelector } from 'react-redux'
 
+import { BaseStackParamList } from 'app/components/app-navigator/types'
 import { TrackTileProps } from 'app/components/track-tile/types'
+import { useDispatchWeb } from 'app/hooks/useDispatchWeb'
+// import { usePushRouteWeb } from 'app/hooks/usePushRouteWeb'
+import { useSelectorWeb } from 'app/hooks/useSelectorWeb'
 import { useThemedStyles } from 'app/hooks/useThemedStyles'
-import { flexCol, flexRow } from 'app/styles'
+import { getPlaying, getPlayingUid } from 'app/store/audio/selectors'
 import { ThemeColors } from 'app/utils/theme'
 
 import { TrackBannerIcon, TrackBannerIconType } from './TrackBannerIcon'
 import { TrackTileBottomButtons } from './TrackTileBottomButtons'
 import { TrackTileCoSign } from './TrackTileCoSign'
+import { TrackTileContainer } from './TrackTileContainer'
 import { TrackTileMetadata } from './TrackTileMetadata'
 import { TrackTileStats } from './TrackTileStats'
 import { TrackTileTopRight } from './TrackTileTopRight'
 
 const createStyles = (themeColors: ThemeColors) =>
   StyleSheet.create({
-    container: {
-      ...flexRow(),
-      minHeight: 152,
-      borderColor: themeColors.neutralLight8,
-      backgroundColor: themeColors.white,
-      borderWidth: 1,
-      borderRadius: 8,
-      maxWidth: 400,
-      marginHorizontal: 'auto',
-      marginBottom: 12,
-      elevation: 3,
-      shadowOpacity: 0.15,
-      shadowOffset: { width: 0, height: 1 },
-      shadowRadius: 3
-    },
     mainContent: {
-      ...flexCol(),
       flex: 1
     }
   })
 
-export const TrackTile = ({
-  artistName,
-  coSign,
-  coverArtSizes,
-  duration,
-  fieldVisibility,
-  goToTrackPage,
-  goToArtistPage,
-  hasCurrentUserReposted,
-  hasCurrentUserSaved,
-  id,
+export const TrackTile = (props: TrackTileProps) => {
+  const { uid } = props
+
+  // Using isEqual as the equality function to prevent rerenders due to object references
+  // not being preserved when syncing redux state from client.
+  // This can be removed when no longer dependent on web client
```

Comment:
Maybe we should come up with a common AUD-XYZ ticket to capture this work so we know where we left all these todos

Reply:
I have a ticket for improving the equality checks: https://linear.app/audius/issue/AUD-1364/reduce-useselectorweb-rerenders

And then the work of moving from useSelectorWeb -> useSelector should be a fairly straighforward find/replace